sdk/servicebus/service-bus/test/serviceBusClient.spec.ts

```diff
@@ -373,32 +372,38 @@ describe("Test createFromAadTokenCredentials", function(): void {
     void
   > {
     const env = getEnvVars();
-
-    let tokenCreds = await loginWithServicePrincipalSecret(
-      env.AAD_CLIENT_ID,
-      env.AAD_CLIENT_SECRET,
-      env.AAD_TENANT_ID,
-      {
-        tokenAudience: aadServiceBusAudience
-      }
-    );
-    sbClient = ServiceBusClient.createFromAadTokenCredentials(123 as any, tokenCreds);
-    should.equal(sbClient.name, "sb://123/", "Name of the namespace is different than expected");
+    if (isNode) {
+      const loginWithServicePrincipalSecret = require("@azure/ms-rest-nodeauth")
+        .loginWithServicePrincipalSecret;
+      let tokenCreds = await loginWithServicePrincipalSecret(
+        env.AAD_CLIENT_ID,
+        env.AAD_CLIENT_SECRET,
+        env.AAD_TENANT_ID,
+        {
+          tokenAudience: aadServiceBusAudience
+        }
+      );
+      sbClient = ServiceBusClient.createFromAadTokenCredentials(123 as any, tokenCreds);
+      should.equal(sbClient.name, "sb://123/", "Name of the namespace is different than expected");
+    }
   });
 
   it("sends a message to the ServiceBus entity", async function(): Promise<void> {
     const env = getEnvVars();
-
-    let tokenCreds = await loginWithServicePrincipalSecret(
-      env.AAD_CLIENT_ID,
-      env.AAD_CLIENT_SECRET,
-      env.AAD_TENANT_ID,
-      {
-        tokenAudience: aadServiceBusAudience
-      }
-    );
-    await testCreateFromAadTokenCredentials(serviceBusEndpoint, tokenCreds);
-    await sbClient.close();
+    if (isNode) {
```

Comment:
What are your thoughts on wrapping the `it` blocks in an `isNode` check rather than performing the check inside the test? It looks like some of these tests should only run in node.js, so we could exclude them from the browser altogether (I think rollup will remove tests wrapped in `if (isNode)`.

Reply:
Since the tests don't get bundled as such, would it add any value?
The mocha's #RunInBrowser hashtag filters out the tests that specifically need to run in browser, so there isn't any runtime optimization either afaik

Reply:
> Since the tests don't get bundled as such

I'm not sure I understand. Doesn't the bundling only remove code that's inside the `if (isNode)` code blocks? If that's the case, then you'll still have these tests in the browser bundle, they'll just be mostly empty (at which point, why not have them removed entirely in the browser vs having a near empty test?)

Reply:
I meant the tests don't get bundled into the artifacts for the end users. So there isn't any value/benefit to the users, nor any test run times?
Sure, the final artifact (browser test code bundling output) can have the Node tests be removed from the file - but I don't see any particular benefit as the main objective here is for the browser bundled code to just not include Node specific dependencies. 

As of today, we don't exclude Node tests from being bundled into the browser output - achieving that would require more efforts and we should likely file a issue. Since test refactoring for Event Hubs and Service Bus is in ongoing plan, it can be looked into then I think.

Reply:
The concern for me is mainly around having the conditional require statements sprinkled around the code. I agree that the ideal solution would need more refactoring, but I believe a midway is possible to achieve.

Since the conditional require is mainly to get the tokenCreds, how about moving the setting of the tokenCreds to `testCreateFromAadTokenCredentials()`?
- The second parameter to `testCreateFromAadTokenCredentials()` which is tokenCreds is made optional
- When no tokenCreds is provided and isNode is true, we require `ms-rest-nodeauth` and create valid tokenCreds inside `testCreateFromAadTokenCredentials()`
- Update the 2 test cases that need valid creds to pass no creds to `testCreateFromAadTokenCredentials()`
- Update the test case that passes empty string as invalid creds to pass say 123 as all it needs is invalid input

Reply:
Or, we can keep things simple and keep the existing design as is and keep the aadUtils.ts file for browser

Reply:
> Update the test case that passes empty string as invalid creds to pass say 123 as all it needs is invalid input

About this, it needed valid credentials to be passed in though.
Updated to refactor and use a `getTokenCredentialsFromAAD` helper instead

Reply:
> Or, we can keep things simple and keep the existing design as is and keep the aadUtils.ts file for browser

Do you mean a separate .browser file for the tests? Introducing a aadUtils.browser file to the source/helper code wouldn't affect the selection of tests / test code to being bundled

Reply:
Updated - I like the `aadUtils.browser` idea as that seems necessary as AAD specifically isn't supported in browser. Moved test resources CRUD operations to `managementUtils.ts`